filters/canada/ontario/kitchener/public-art-geojson-to-json.js

```diff
@@ -1,40 +1,94 @@
-const filter = function (data, std_lib, stringify, skip_errors) {
+const filter = function (data, std_lib, schema, validator, stringify) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let remove_if_empty = std_lib.get("remove_if_empty");
+    let validate_params = std_lib.get("validate_params");
+    let remove_null_date_fields = std_lib.get("remove_null_date_fields");
+    let create_dates_template = std_lib.get("create_dates_template");
+    
+    // validate parameters object
+    schema = validate_params(schema, validator);
+
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
     
     let new_data = [];
+    let errors = [];
 
     data["features"].map(d => {
         let item = {};
-        item.name = d.properties.MAP_LABEL;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+        
+        // add name (required)
+        if (!add_required(item, "name", d, d.properties.MAP_LABEL, errors)) {
+            skip = true;
+        }
+
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.geometry.coordinates[0], errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: d.geometry.coordinates[0], latitude: d.geometry.coordinates[1]};
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, d.geometry.coordinates[1], errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        let details = {};
-        if (d.properties["Description"] != null || d.properties["Description"] != undefined) {
-            details.description = d.properties["Description"];
-        }
-        details.artist_first_name = d.properties["ARTIST_FIRST_NAME"];
-        details.artist_last_name = d.properties["ARTIST_LAST_NAME"];
-        details.location = d.properties["LOCATION"];
-        details.location_description = d.properties["LOCATION_DESCRIPTION"];
+        //optional fields
+        // item.artist = d.properties.ARTIST_FIRST_NAME + " " + d.properties.ARTIST_LAST_NAME
+        let artist_name;
+        if (d.properties.ARTIST_FIRST_NAME != null) artist_name = d.properties.ARTIST_FIRST_NAME;
+        if (d.properties.ARTIST_LAST_NAME != null) artist_name += " " + d.properties.ARTIST_LAST_NAME;
```

Comment:
you may want to check if the artist's first and last name are empty strings ("") or undefined as well..